tests/torch/sparsity/movement/test_model_saving.py

```diff
@@ -109,6 +121,145 @@ def test_same_outputs_in_torch_and_exported_onnx(self, tmp_path: Path, recipe: B
         assert np.allclose(softmax(onnx_outputs, axis=-1),
                            softmax(torch_outputs, axis=-1), atol=1e-6)
 
+    @pytest.mark.skipif(parse_version(torch.__version__) < parse_version("1.12"),
+                    reason=f"torch {torch.__version__} is not compatible with installed transformers package. "
+                            f"Some tests may fail with segmentation fault")
+    @pytest.mark.parametrize('desc', [
+        Dict(
+            nncf_weight_ratio=0.43,
+            ov_weight_ratio=0.36,
+            recipe=BertRunRecipe().model_config_(
+                max_position_embeddings=2,
+                intermediate_size=4,
+                vocab_size=2,
+                num_labels=1,
+                hidden_size=4,
+                num_attention_heads=4,
+                num_hidden_layers=1,
+            ),
+        ),
+        Dict(
+            nncf_weight_ratio=0.31,
+            ov_weight_ratio=0.25,
+            recipe=Wav2Vec2RunRecipe().model_config_(
+                intermediate_size=4,
+                num_labels=1,
+                num_conv_pos_embeddings=1,
+                num_attention_heads=4,
+                classifier_proj_size=1
+            )
+        ),
+        Dict(
+            nncf_weight_ratio=0.41,
+            ov_weight_ratio=0.07,
+            recipe=Wav2Vec2RunRecipe().model_config_(
+                # stride, hidden size and num heads is selected to reproduce invalid parsing of reshape:
+                # input_shape=[1, 4, 1, 2] and output shape = [4, 1, 2]. NGraph transformation eager finds extend and
+                # shrink mode in the reshape at the same time:
+                # [1, 4, 1] is mapped to [4] and [2] - to [1,2]. Such reshape is skipped from pruning and heads are not
+                # removed. But this reshape is just unsqueeze and pruning should work fine.
+                conv_stride=(5, 5),
+                hidden_size=8,
+                num_attention_heads=4,
+                # parameters below are for higher compression rate in NNCF: less parameters for not-pruned parts.
+                intermediate_size=4,
+                num_labels=1,
+                num_conv_pos_embeddings=1,
+                classifier_proj_size=3,
+            )
+        ),
+        Dict(
+            nncf_weight_ratio=0.43,
+            ov_weight_ratio=0.29,
+            recipe=SwinRunRecipe().model_config_(
+                num_heads=[4],
+                num_labels=1,
+                num_channels=4,
+                window_size=1,
+            ),
+        ),
+        Dict(
+            nncf_weight_ratio=0.55,
+            ov_weight_ratio=0.20,
+            recipe=DistilBertRunRecipe().model_config_(),
+        ),
+        Dict(
+            nncf_weight_ratio=0.63,
+            ov_weight_ratio=0.38,
+            recipe=MobileBertRunRecipe().model_config_(),
```

Comment:
Big difference for MobileBERT reveals some bug with pruning by row and column at the same time.
![MicrosoftTeams-image (6)](https://user-images.githubusercontent.com/4014476/233698118-5a8c4cab-5a6b-40af-a499-0549de6fde6d.png)